### PR TITLE
Update Mock Builder for Post-Capella Tests

### PR DESCRIPTION
**Pull request by ethDreamer**
Updated the mock builder and also added some Capella tests and validation of the withdrawals_root for builder payloads. We need to merge:

* #3944 

before merging this.

### CODE REVIEW

File: beacon_node/execution_layer/src/lib.rs [hunk 1]
```diff
@@ -1965,6 +1970,13 @@ fn verify_builder_bid<T: EthSpec, Payload: AbstractExecPayload<T>>(
         );
     }
 
+    let expected_withdrawals_root = payload_attributes
+        .withdrawals()
+        .ok()
+        .cloned()
+        .map(|withdrawals| Withdrawals::<T>::from(withdrawals).tree_hash_root());
```
Comment (realbigsean) [inline beacon_node/execution_layer/src/lib.rs:1977]: we can use the withdrawals root from the local payload header here and fallback to re-calculating the root when we don't have a header

File: beacon_node/execution_layer/src/test_utils/mock_builder.rs [hunk 2]
```diff
@@ -39,25 +43,129 @@ pub enum Operation {
...
+        &mut self,
+        signing_key: &SecretKey,
+        context: &Context,
+    ) -> Result<BlsSignature, Error>;
+
+    fn to_signed_bid(self, signature: BlsSignature) -> SignedBuilderBid;
+}
+
+impl BidStuff for BuilderBid {
+    fn fee_recipient(&mut self) -> &mut ExecutionAddress {
```
Comment (realbigsean) [inline beacon_node/execution_layer/src/test_utils/mock_builder.rs]: would you mind renaming these to e.g. `fee_recipient_mut()`

File: beacon_node/execution_layer/src/lib.rs (hunk 1 above)
Comment (ethDreamer) [reply beacon_node/execution_layer/src/lib.rs:1977]: I calculate both here because the point is to compare them to ensure they match.

Comment (realbigsean) [reply beacon_node/execution_layer/src/lib.rs:1977]: yea was talking about local payload withdrawals root (not payload attributes) root vs the relay withdrawals root. but as you pointed out offline doing it this way avoids misconstruing an issue with the local EE vs an issue with the builder

File: beacon_node/execution_layer/src/test_utils/mock_builder.rs (hunk 2 above)
Comment (ethDreamer) [reply beacon_node/execution_layer/src/test_utils/mock_builder.rs]: done!

**Review by realbigsean — APPROVED** [review]

**Pull request merged**